client/src/utilities/chart.js

```diff
@@ -195,6 +198,7 @@ export function getPieData(visualisation, datasets) {
   const { datasetId, spec } = visualisation;
   const dataset = datasets[datasetId];
   const bucketIndex = getColumnIndex(dataset, spec.bucketColumn);
+  const bucketColumnType = dataset.get('columns').get(bucketIndex).get('type');
```

Comment:
Instead of chained `get`s you can also use `getIn`: dataset.getIn(['columns', bucketIndex, 'type']);